addons/account_edi/models/account_move.py

```diff
@@ -135,6 +135,66 @@ def _compute_edi_show_abandon_cancel_button(self):
     # Export Electronic Document
     ####################################################
 
+    def _prepare_edi_vals_to_export(self):
+        ''' The purpose of this helper is to prepare values in order to export an invoice through the EDI system.
+        This includes the computation of the tax details for each invoice line that could be very difficult to
+        handle regarding the computation of the base amount.
+        :return: A python dict containing default pre-processed values.
+        '''
+        self.ensure_one()
+
+        res = {
+            'record': self,
+            'invoice_line_vals_list': [],
+        }
+
+        # Invoice lines details.
+        tax_detail_per_tax = {}
+        added_base_amount_keys = set()
+        for index, line in enumerate(self.invoice_line_ids.filtered(lambda line: not line.display_type), start=1):
+            line_vals = {
+                'index': index,
+                **line._prepare_edi_vals_to_export(),
+            }
+            res['invoice_line_vals_list'].append(line_vals)
+
+            # Tax details.
+            for tax_vals in line_vals['tax_detail_vals_list']:
+                tax_detail_per_tax.setdefault(tax_vals['tax'], {
+                    'tax': tax_vals['tax'],
+                    'src_tax': tax_vals['src_tax'],
+                    'tax_base_amount_currency': 0.0,
+                    'tax_amount_currency': 0.0,
+                    'tag_ids': set(),
+                })
+                vals = tax_detail_per_tax[tax_vals['tax']]
+
+                # Avoid adding multiple times the same base (e.g. with multiple taxes on the same line).
```

Comment:
What about bases that can change?  We don't change the base, but the calculation of the subsequent tax?  (need to see for based on base amount/ base amount + VAT / only VAT / ...  

I don't get however how you can have twice the same tax on an invoice line.

Reply:
You can have multiple tax repartition lines on the same tax

Reply:
But don't we have the tax_detail_per_tax already per tax?